src/shape.cpp

```diff
@@ -247,6 +247,26 @@ Dimension Dimension::any() {
     return Dimension();
 }
 
+dimension_value_t Dimension::getMinValueRegardless() const {  // TODO throw if any
+    return isStatic() ? getStaticValue() : getMinValue();
+}
+
+dimension_value_t Dimension::getMaxValueRegardless() const {
+    return isStatic() ? getStaticValue() : getMaxValue();
+}
+
+std::optional<Dimension> Dimension::createIntersection(const Dimension& other) const {
+    if (*this == Dimension::any())
+        return other;
+    if (other == Dimension::any())
+        return *this;
+    auto start = std::max(this->getMinValueRegardless(), other.getMinValueRegardless());
+    auto end = std::min(this->getMaxValueRegardless(), other.getMaxValueRegardless());
+    if (end < start)
+        return std::nullopt;
+    return Dimension{start, end};
```

Comment:
We should not create dimension with constructor Dimension(X, Y) when X == Y. This only works because we did not finish this task: https://jira.devtools.intel.com/browse/CVS-75096

Please use static dim constructor if the intersection is static:
```
return start == end ? Dimension{start} : Dimension{start, end};
```

Reply:
I don't think the issue is with "which constructor is called here". Why DImension(2,2) should represent something different than DImension(2)? I do agree that in test I should probably avoid directly using Dimension(2,2) if it is static for readability purpose but otherwise for me it is perfectly correct constructor. It is up to the Dimension to decide if it is static if not, and simplifies code here.